### PR TITLE
Add cv_timedwait_sig_hires to allow interruptible sleep

### PR DESCRIPTION
**Pull request by tuxoko**
Signed-off-by: Chunwei Chen <david.chen@osnexus.com>

### CODE REVIEW

File: module/spl/spl-condvar.c [hunk 1]
```diff
@@ -312,10 +312,27 @@ cv_timedwait_hires(kcondvar_t *cvp, kmutex_t *mp, hrtime_t tim, hrtime_t res,
...
+cv_timedwait_hires(kcondvar_t *cvp, kmutex_t *mp, hrtime_t tim, hrtime_t res,
+    int flag)
+{
+	return (cv_timedwait_hires_common(cvp, mp, tim, res, flag,
+	    TASK_UNINTERRUPTIBLE));
 }
 EXPORT_SYMBOL(cv_timedwait_hires);
 
+hrtime_t
+cv_timedwait_hires_sig(kcondvar_t *cvp, kmutex_t *mp, hrtime_t tim, hrtime_t res,
```
Comment (behlendorf) [inline module/spl/spl-condvar.c]: We should call this `cv_timedwait_sig_hires()` which is consistent with what it's called in [illumos](https://github.com/openzfs/openzfs/blob/master/usr/src/uts/common/os/condvar.c#L363).

Comment (tuxoko) [inline module/spl/spl-condvar.c]: done.

File: module/spl/spl-condvar.c [hunk 2]
```diff
@@ -231,7 +231,7 @@ EXPORT_SYMBOL(__cv_timedwait_sig);
  * 'expire_time' argument is an absolute clock time in nanoseconds.
  * Return value is time left (expire_time - now) or -1 if timeout occurred.
  */
-static clock_t
+static hrtime_t
 __cv_timedwait_hires(kcondvar_t *cvp, kmutex_t *mp, hrtime_t expire_time,
```
Comment (behlendorf) [inline module/spl/spl-condvar.c]: This bugs me too but the illumos version of this does return a `clock_t`.  However, it looks like have a longer standing bug where we're not converting the hrtime_t to a clock_t with `NSEC_TO_TICK().`

Comment (tuxoko) [inline module/spl/spl-condvar.c]: We don't seems to use return value anyway.

Comment (behlendorf) [inline module/spl/spl-condvar.c]: So at the moment it's harmless but let's preserve the existing upstream prototype, and even better for the return value.

Comment (tuxoko) [inline module/spl/spl-condvar.c]: Right.

**Comment by behlendorf** [comment]
LGTM

**Comment by tuxoko** [comment]
Change usleep_range to schedule_hrtimeout_range

**Pull request closed without merging**